crates/dojo-core/src/auth/systems.cairo

```diff
@@ -34,22 +34,20 @@ mod RouteAuth {
 
         // Set status
         commands::set_entity(
-            (route.role_id, route.resource_id).into(),
-            (AuthStatus { is_authorized: true })
+            (route.role_id, route.resource_id).into(), (AuthStatus { is_authorized: true })
         );
     }
 }
 
 #[system]
 mod IsAccountAdmin {
     use traits::Into;
-    use starknet::get_tx_info;
     use box::BoxTrait;
     use dojo_core::{auth::components::{AuthStatus, AuthRole}, integer::u250};
 
     fn execute() -> bool {
         // Get calling account contract address
-        let caller = get_tx_info().unbox().account_contract_address; // tx origin
+        let caller = ctx.caller_account;
```

Comment:
i think let's pass in context here explicitly. when we do the codegen and it isn't passed in explicitly, we can compile it to `execute(_: Context)`. otherwise it is a bit too magical

we should also enforce that it is the first parameter

Reply:
That's a good idea! I've updated this in the compiler to make it more explicit